### PR TITLE
implement get /api/categories endpoint

### PR DESCRIPTION
**Pull request by impemberton**

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -0,0 +1,32 @@
+const request = require("supertest");
+const app = require("../app.js");
+const seed = require("../db/seeds/seed");
+const testData = require("../db/data/test-data");
+
+beforeEach(() => seed(testData));
+
```
Comment (callumu21) [inline __tests__/app.test.js:9]: Could you add an afterAll function here that ends the connection? Check NC Notes if you need a reminder.

File: __tests__/app.test.js [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+describe("/api/categories", () => {
+  test("GET - 200 responds with an array of category objects", () => {
+    return request(app)
+        .get("/api/categories")
+        .expect(200)
+        .then(res => {
+            expect(Array.isArray(res.body)).toBe(true);
+            res.body.forEach(category => {
```
Comment (callumu21) [inline __tests__/app.test.js:18]: Before this .forEach, you need to check the length of the array that's been returned. If it's a length of 0, you won't get into this block and get an automatic pass.

File: app.js [hunk 3]
```diff
@@ -0,0 +1,13 @@
+const { getCategories } = require("./controllers/controllers");
+const express = require("express");
+const app = express();
+
+app.use(express.json());
```
Comment (callumu21) [inline app.js]: This should be removed. You won't need this until you're processing the request body.

File: controllers/controllers.js [hunk 4]
```diff
@@ -0,0 +1,9 @@
+const { selectCategories } = require("../models/models");
```
Comment (callumu21) [inline controllers/controllers.js]: You might want to rename your current model and categories file to be more semantic (e.g. categoriesController/categoriesModel). Not an issue now, but something to watch out for later.

**Review by callumu21 — CHANGES_REQUESTED** [review]
Hi Isaac. Just a few changes to make, please. If you could send a new PR link to us to check over before merging, that would be great. Other than that, great start 👍

**Comment by Geo-Beard** [comment]
Looks good! Feel free to merge :)

**Pull request merged**